### PR TITLE
Publish NotifierNotFound in traits.observation.api

### LINKED ISSUE #1479: `NotifierNotFound` from traits.observation is not in the public API
The `NotifierNotFound` exception can be raised when trying to disconnect an observer which is not connected.  However it is not public so there is no way to catch it without importing from private modules.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR makes `NotifierNotFound` importable from `traits.observation.api`, updates the documentation to note that `NotifierNotFound` can be raised, and adds a test for `NotifierNotFound` being raised.

It also renames the `traits.observation._exception` module to `traits.observation.exception`, in keeping with that package's conventions for public versus private objects.

Closes #1479.

**Checklist**
- [x] Tests
- [x] Update API reference (`docs/source/traits_api_reference`)
- [x] Update User manual (`docs/source/traits_user_manual`)
- ~[ ] Update type annotation hints in `traits-stubs`~ N/A

### CODE REVIEW

File: traits/observation/api.py [hunk 1]
```diff
@@ -8,12 +8,14 @@
 #
 # Thanks for using Enthought open source!
 
-from traits.observation.exception_handling import (     # noqa: F401
```
Comment (mdickinson) [inline traits/observation/api.py:11]: The '# noqa's here are unnecessary - they're taken care of by the `per-file-ignores` setting in `setup.cfg`.

Comment (mdickinson) [reply traits/observation/api.py]: I'll revert here and make these changes in a separate PR.

**Review by corranwebster — APPROVED** [review]
Only concern is whether we should keep a dummy `_exceptions` module that imports from the new `exceptions` to support code that had to go rummaging around in private modules to get the exception objects.

I'm in two minds: it was private, so code doing this knew what it was getting into; on the other hand it was occasionally necessary due to the issue with the API, so helping code that was forced to do that might be polite.

**Comment by mdickinson** [comment]
> Only concern is whether we should keep a dummy `_exceptions` module that imports from the new `exceptions`

I think we should be fine without it. A search turned up no uses of `NotifierNotFound` in other public or private projects that I have access to.

**Review by rahulporuri — APPROVED** [review]

**Pull request merged**